cli/tests/unit/tls_test.ts

```diff
@@ -232,3 +233,41 @@ unitTest(
     conn.close();
   },
 );
+unitTest(
+  { perms: { net: true, read: true } },
+  async function getAndConnectWithSNI(): Promise<void> {
+    const resolvable = deferred();
+    const hostname = "localhost";
+    const port = 3600;
+    const certFile = "cli/tests/tls/localhost.crt";
+    const keyFile = "cli/tests/tls/localhost.key";
+    const rootCertFile = "cli/tests/tls/RootCa.pem";
+
+    const listener = Deno.listenTls({
+      hostname,
+      port,
+      certFile,
+      keyFile,
+    });
+
+    listener.accept().then((conn) => {
+      assertEquals((conn as Deno.Conn & { sni?: string }).sni, hostname);
+      setTimeout(() => {
+        listener.close();
+        conn.close();
+        resolvable.resolve();
+      }, 0);
+    });
+
+    async function connect() {
+      const conn = await Deno.connectTls({
+        hostname,
+        port,
+        certFile: rootCertFile,
+      });
+      conn.close();
+    }
+
+    await resolvable;
```

Comment:
I think this test can be written without a `deferred`.

Also it looks like the `connect` function is never called right? It looks to me like this test should just be hanging forever.

```suggestion
  async function getAndConnectWithSNI(): Promise<void> {
    const hostname = "localhost";
    const port = 3600;
    const certFile = "cli/tests/tls/localhost.crt";
    const keyFile = "cli/tests/tls/localhost.key";
    const rootCertFile = "cli/tests/tls/RootCa.pem";

    const listener = Deno.listenTls({
      hostname,
      port,
      certFile,
      keyFile,
    });

    const incomingPromise = listener.accept();
    
    const outgoing = await Deno.connectTls({
      hostname,
      port,
      certFile: rootCertFile,
    });
    outgoing.close();
    
    const incoming = await incomingPromise;
    assertEquals((incoming as Deno.Conn & { sni?: string }).sni, hostname);
    incoming.close();
    listener.close();
```

Reply:
Yes, you're right! Don't know why my changes weren't pushed, the tests worked locally before pushing. I'm digging into it.